### ISSUE TITLE
[CW-2848] Feature: Trigger Outgoing Messages based on events

### ISSUE THREAD

**Issue by sojan-official** [chatwoot/chatwoot#657]
**Is your feature request related to a problem? Please describe.**
Currently, chatwoot conversations are only triggered when a user initiates the conversation. I would like to have the ability to configure outgoing Messages in chatwoot based on Events.

**Describe the solution you'd like**
-  Ability to pre-configure outgoing messages in chatwoot for specific events 
- the events could be  `user's first visit`, `user visiting nth time` , `user visiting the specific page`, `user closing chat window with out typing` etc

**Describe alternatives you've considered**
none

<sub>[CW-2848](https://linear.app/chatwoot/issue/CW-2848/feature-trigger-outgoing-messages-based-on-events)</sub>

**Comment by amesas** [comment]
This is a must to compete with intercom!

**Comment by feconroses** [comment]
This feature would be super helpful. Are there any plans to work on this?

**Comment by pranavrajs** [comment]
@feconroses Yes, this is in the works. We have released a basic outgoing campaign (#2227) in the last release with an option to set a campaign based on the URL and the time on the page.
